pkg/api/norman/customization/cluster/actions_kubeconfig.go

```diff
@@ -64,6 +64,9 @@ func (a ActionHandler) GenerateKubeconfigActionHandler(actionName string, action
 		}
 
 		cfg, err = kubeconfig.ForClusterTokenBased(&cluster, apiContext.ID, apiContext.Request.Host, tokenKey)
+		if err != nil {
+			return err
+		}
```

Comment:
An error check is already being done on L73.

Reply:
The `err` variable is being shadowed on line 46, so the value of `err` on line 73 is different.

Reply:
Trying to hold that error all the way through is a bad pattern, it takes too long to parse this function to determine where those errors are being set and as Donnie found, we can accidentally shadow it and never catch the error. We should be checking them when we create them.